core/state/state_object.go

```diff
@@ -395,6 +395,63 @@ func (s *stateObject) CommitTrie(db Database) (*trie.NodeSet, error) {
 	return nodes, err
 }
 
+// DeleteTrie the storage trie of the object from db.
+func (s *stateObject) DeleteTrie(db Database) (*trie.NodeSet, error) {
```

Comment:
This method name seems very ambiguous to me. I'd expect this to actually delete something, especially given the passed in db. The method comment also doesn't make much sense.

Reply:
The comment for CommitTrie is also very strange